apps/src/templates/FinishDialog.jsx

```diff
@@ -0,0 +1,299 @@
+    margin: 10,
+    verticalAlign: 'middle',
+  },
+  blockCountPerfect: {
+    color: color.level_perfect,
+  },
+  blockCountPass: {
+    color: color.light_teal,
+  },
+  blockCountDescriptor: {
+    borderWidth: 1,
+    borderStyle: 'solid',
+    fontFamily: '"Gotham 5r", sans-serif',
+    borderRadius: 5,
+    padding: 5,
+    verticalAlign: 'middle',
+  },
+  achievements: {
+    width: 217,
+    display: 'block',
+    margin: '14px 0px 0px',
+    padding: '0px 23px',
+    borderColor: color.light_teal,
+    borderWidth: 1,
+    borderStyle: 'solid',
+    borderRadius: 5,
+    fontFamily: '"Gotham 5r", sans-serif',
+    color: color.dark_charcoal,
+  },
+  achievementIcon: {
+    width: 16,
+  },
+  achievementRow: {
+    margin: 4,
+  },
+  funometer: {
+    marginTop: 5,
+    display: 'flex',
+    flexDirection: 'row-reverse',
+  },
+  button: {
+    borderWidth: 0,
+    height: 40,
+    color: color.white,
+    margin: '0px 5px',
+  },
+  replayButton: {
+    backgroundColor: color.green,
+  },
+  continueButton: {
+    backgroundColor: color.orange,
+  },
+};
+
+export class UnconnectedFinishDialog extends Component {
+  static propTypes = {
+    isOpen: PropTypes.bool,
+    hideBackdrop: PropTypes.bool,
+
+    onContinue: PropTypes.func,
+    onReplay: PropTypes.func,
+
+    isPerfect: PropTypes.bool,
+    blocksUsed: PropTypes.number,
+    blockLimit: PropTypes.number,
+    achievements: PropTypes.arrayOf(PropTypes.shape({
+      isAchieved: PropTypes.bool,
+      iconUrl: PropTypes.string,
+      message: PropTypes.string,
+    })),
+    showFunometer: PropTypes.bool,
+    canShare: PropTypes.bool,
+  };
+
+  getBubble() {
+    return (
+      <div style={styles.bubbleContainer}>
+        <div
+          style={{
+            ...styles.bubble,
+            backgroundColor:
+              this.props.isPerfect ? color.level_perfect : color.level_passed,
+          }}
+        />
+      </div>
+    );
+  }
+
+  getBlockCountDescription() {
+    if (this.props.blocksUsed < this.props.blockLimit) {
+      return msg.betterThanPerfectDescription();
+    }
+    if (this.props.blocksUsed === this.props.blockLimit) {
+      return msg.perfectDescription();
+    }
+    if (this.props.blocksUsed > this.props.blockLimit) {
+      return msg.tooManyBlocksDescription();
+    }
+  }
+
+  getBlockCounter() {
+    if (this.props.blockLimit === undefined) {
+      return null;
+    }
+
+    const tooManyBlocks = this.props.blocksUsed > this.props.blockLimit;
+    return (
+      <div style={styles.blockCountWrapper}>
+        <span style={styles.blockCountLabel}>
+          {msg.numBlocksUsedLabel()}:
+        </span>
+        <span
+          style={tooManyBlocks ?
+            styles.blockCountPass : styles.blockCountPerfect}
+        >
+          <span style={styles.blockCount}>
+            {(this.props.blocksUsed || 0).toString()}
+            {this.props.blockLimit && ('/' + this.props.blockLimit.toString())}
```

Comment:
This looks fine to ship, but will we need to re-visit for RTL?

Reply:
Probably. Fractions in Arabic still have the numerator on the left, but it might differ in other RtL languages (or have some other issue in Arabic).